### ISSUE TITLE
nginx naxsi rule fails when using regex

### ISSUE THREAD

**Issue by 8191** [opnsense/plugins#3497]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

**Describe the bug**
When using a Naxsi rule which has _Use Regular Expressions_ checked, the nginx config validation fails.

**To Reproduce**
1. Go to HTTP(S) > Naxsi WAF Rule
2. Add a rule which has _Use Regular Expressions_ checked
3. Enter any value in _Match Value_ (a single ASCII character is sufficient)
4. Use the rule in a Naxsi Policy
5. Assign the policy to a location
6. Assign the location to a HTTP server instance
7. Restart the Nginx server

**Expected behavior**
Nginx server should restart successfully

**Relevant log files**
```
<15>1 2023-07-07T22:17:05+02:00 waf-prod.dmz.faux.at nginx 28411 - [meta sequenceId="7"] NGINX setup routine started.
<8>1 2023-07-07T22:17:06+02:00 waf-prod.dmz.faux.at nginx 28411 - [meta sequenceId="8"] nginx: [emerg] Naxsi-Config : Incorrect line MainRule id:1000 (/usr/obj/usr/ports/www/nginx/work/naxsi-29793dc/naxsi_src/naxsi_skeleton.c/973)... in /usr/local/etc/nginx/nginx.conf:64
<8>1 2023-07-07T22:17:06+02:00 waf-prod.dmz.faux.at nginx 28411 - [meta sequenceId="9"] nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed
```

The rule itself looks fine:
```
MainRule id:1000 "rx:select|union|update|delete|insert|table|from|ascii|hex|unhex|drop|load_file|substr|group_concat|dumpfile" "msg:sql keywords" "mz:BODY|ARGS|URL|$HEADERS_VAR_X:Cookie" "s:$policy425ddd58c0d046638ef0d0c4bfb20be1:4";
```

**Additional context**
In previous versions (e.g. 23.1.1) the rule worked fine.

Simply unchecking the _Use regular expression_ checkbox solves the issue; the line then looks as follows:
```
MainRule id:1000 "str:select|union|update|delete|insert|table|from|ascii|hex|unhex|drop|load_file|substr|group_concat|dumpfile" "msg:sql keywords" "mz:BODY|ARGS|URL|$HEADERS_VAR:Cookie" "s:$policy425ddd58c0d046638ef0d0c4bfb20be1:4";
```

Simplifying the regex to e.g. `select` does not work neither.

It seems like regex support is broken in naxsi.

**Environment**
OPNsense 23.1.10_1 (amd64, OpenSSL).

**Comment by 8191** [comment]
Might be related: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=271064

**Comment by fichtner** [comment]
It’s fixed on 23.1.11 already… 😉

**Comment by 8191** [comment]
Fixed with OPNsense 23.1.11

**Comment by 8191** [comment]
> It’s fixed on 23.1.11 already… 😉

Yeah, I was too fast in reporting this and even investigating it, before checking for latest versions... :-)
